pxr/usd/lib/usd/stage.cpp

```diff
@@ -1732,6 +1732,36 @@ UsdStage::GetPrimAtPath(const SdfPath &path) const
     return UsdPrim(primData, proxyPrimPath);
 }
 
+UsdObject
+UsdStage::GetObjectAtPath(const SdfPath &path) const
+{
+    // Maintain consistent behavior with GetPrimAtPath
+    if (!path.IsAbsolutePath()) {
+        return UsdObject();
+    }
+
+    bool isPrimPath = path.IsPrimPath();
+    bool isPropPath = path.IsPropertyPath();
+    if (!isPrimPath && !isPropPath) {
+        return UsdObject();
+    }
+
+    // A valid prim must be found to return either a prim or prop
+    auto prim = GetPrimAtPath(path.GetPrimPath());
+    if (prim) {
+        if (isPrimPath) {
+            return prim.As<UsdObject>();
```

Comment:
Shouldn't need to explicitly cast to UsdObject using As() here and below, the implicit derived-to-base conversion should be sufficient.